go/types/sequence_iterator.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+// iterSequence efficiently iterates through a sequence calling cb for
+// each item.
+//
+// This is preferred to sequenceCursor.iter()
+func iterSequence(sc *sequenceCursor, cb cursorIterCallback) {
+	cur := sc.clone()
+	cur.enableReadAhead()
+	it := &sequenceIterator{cur}
+	for it.hasMore() && !cb(it.item()) {
+		it.advance(1)
+	}
+}
+
+// sequenceIterator iterates forward through a sequence.
+//
+// Since it can assume the sequence is being read forward, it reads
+// upcoming chunks ahead of their access to optimize throughput
+type sequenceIterator struct {
+	cursor *sequenceCursor
+}
+
+// newSequenceIterator creates an iterator for reading a sequence
+func newSequenceIterator(seq sequence, idx uint64) *sequenceIterator {
+	sc := newCursorAtIndex(seq, idx)
+	sc.enableReadAhead()
+	return &sequenceIterator{sc}
+}
+
+
+// hasMore return true if there's more to iterate
+func (si sequenceIterator) hasMore() bool {
```

Comment:
What do you think about adhering to the Set/ListIterator style of interface, and only providing a `Next()` method, which returns a Value or nil if there are no more values?